### DISCUSSION TITLE
docker安装v2ray无法连接

### DISCUSSION THREAD

**Discussion by lnermif** [v2fly/v2ray-core#546] [Q&A]
1) 你正在使用哪个版本的 V2Ray？
V2ray 4.33.0

2) 你的使用场景是什么？比如使用 Chrome 通过 Socks/VMess 代理观看 YouTube 视频。
使用v2rayN通过VMess代理上网

3) 你看到的不正常的现象是什么？（请描述具体现象，比如访问超时，TLS 证书错误等）
V2rayN客户端无法连接服务端

4) 你期待看到的正确表现是怎样的？
能连接上

5) 请附上你的配置（提交 Issue 前请隐藏服务器端IP地址）。

服务器端配置:

```javascript
{
	"log": {
		"access": "/var/log/v2ray/access.log",
		"error": "/var/log/v2ray/error.log",
		"loglevel": "warning"
	},
	"inbounds": [{
		"port": 14424,
		"listen": "127.0.0.1",
		"protocol": "vmess",
		"settings": {
			"clients": [{
				"id": "",
				"level": 1,
				"email": "love@v2fly.org"
			}],
			"decryption": "none"
		},
		"streamSettings": {
			"network": "ws",
			"security": "none",
			"wsSettings": {
				"path": "/ray"
			}
		}
	}],
	"outbounds": [{
		"protocol": "freedom",
		"settings": {}
	}]
}
```

客户端配置:

```javascript
{
	"outbounds": [{
			"tag": "proxy",
			"protocol": "vmess",
			"settings": {
				"vnext": [{
					"address": "域名",
					"port": 443,
					"users": [{
						"id": "",
						"alterId": 32,
						"email": "t@t.tt",
						"security": "auto"
					}]
				}]
			},
			"streamSettings": {
				"network": "ws",
				"wsSettings": {
					"path": "/ray"
				}
			},
			"mux": {
				"enabled": true,
				"concurrency": 8
			}
		},
		{
			"tag": "direct",
			"protocol": "freedom",
			"settings": {}
		},
		{
			"tag": "block",
			"protocol": "blackhole",
			"settings": {
				"response": {
					"type": "http"
				}
			}
		}
	],
	"stats": {},
	"api": {
		"tag": "api",
		"services": [
			"StatsService"
		]
	},
	"routing": {
		"domainStrategy": "IPIfNonMatch",
		"rules": [{
			"type": "field",
			"inboundTag": [
				"api"
			],
			"outboundTag": "api"
		}]
	}
}

```

6)  请附上出错时软件输出的错误日志。在 Linux 中，日志通常在 `/var/log/v2ray/error.log` 文件中。

服务器端错误日志:

```javascript
2020/12/24 05:23:26 [Warning] v2ray.com/core: V2Ray 4.33.0 started
```

客户端错误日志:

```javascript
2020/12/24 14:35:12 [Warning] failed to handler mux client connection > v2ray.com/core/proxy/vmess/outbound: failed to find an available destination > v2ray.com/core/common/retry: [v2ray.com/core/transport/internet/websocket: failed to dial WebSocket > v2ray.com/core/transport/internet/websocket: failed to dial to (ws://域名:443/ray): 400 Bad Request > websocket: bad handshake] > v2ray.com/core/common/retry: all retry attempts failed
2020/12/24 14:35:12 [Warning] [2508141857] v2ray.com/core/proxy/http: failed to read response from speedtest-sgp1.digitalocean.com > unexpected EOF
```

7) 请附上访问日志。在 Linux 中，日志通常在 `/var/log/v2ray/access.log` 文件中。

```javascript
2020/12/18 17:34:18 ip:63571 rejected  v2ray.com/core/proxy/vmess/encoding: failed to drain connection DrainSize = 3079 33 14 > v2ray.com/core/proxy/vmess/encoding: invalid user > v2ray.com/core/proxy/vmess: Not Found
```

8) 其它相关的配置文件（如 Nginx）和相关日志。
nginx配置 server节点内
location /ray { # 与 V2Ray 配置中的 path 保持一致
      if ($http_upgrade != "websocket") { # WebSocket协商失败时返回404
          return 404;
      }
      proxy_redirect off;
      proxy_pass http://127.0.0.1:14424; # 假设WebSocket监听在环回地址的10000端口上
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_set_header Host $host;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    }

9) 如果 V2Ray 无法启动，请附上 `--test` 输出。
docker 容器列表已显示启动

10) 其他说明。
服务端使用的是docker安装方式 防火墙端口已经放行14424端口
`docker run -d --name v2ray -v /etc/v2ray:/etc/v2ray -v /var/log/v2ray:/var/log/v2ray -p 14424:14424 v2fly/v2fly-core v2ray -config=/etc/v2ray/config.json`

套用的是CF 域名为二级域名
服务器时区为+8 时间和客服端时间相同

nginx日志
[23/Dec/2020:17:50:13 +0800] "GET /ray HTTP/1.1" 502 150 "-" "Go-http-client/1.1"

**Comment by kslr** [comment]
删掉alterid

**Comment by lnermif** [comment]
> 删掉alterid
删除掉也是同样的问题
